modules/tensor_mechanics/src/actions/PrescribedBCAction.C

```diff
@@ -0,0 +1,190 @@
+/****************************************************************/
+/* MOOSE - Multiphysics Object Oriented Simulation Environment  */
+/*                                                              */
+/*          All contents are licensed under LGPL V2.1           */
+/*             See LICENSE for full restrictions                */
+/****************************************************************/
+#include "PrescribedBCAction.h"
+
+#include "Factory.h"
+#include "FEProblem.h"
+#include "Parser.h"
+#include "Conversion.h"
+
+template<>
+InputParameters validParams<PrescribedBCAction>()
+{
+  InputParameters params = validParams<Action>();
+  params.addClassDescription("Prescribes acceleration or displacement time history at a boundary. If the user provided displacement time history, this is converted to acceleration time history using Backward Euler. This acceleration time history or the user supplied acceleration time history is then converted into displacement using Newmark time integration and this displacement is applied as a FunctionPresetBC.");
+  params.addRequiredParam<NonlinearVariableName>("variable", "variable on which this boundary condition is applied.");
+  params.addRequiredParam<FileName>("data_file", "File name of the file containing displacement/acceleration time history. The file should contain two columns. First column is the time vector starting at 0.0 and the second column is the displacement/acceleration as a function of time. The displacement/acceleration at t = 0.0 should be zero. Also the time step used in the data file should match the time step dt and dtmin provided in the Executioner block for prescribed displacement.");
```

Comment:
I see, you are using finite differencing to build time derivatives... Unfortunately you seem to need a second derivative here, `Function` objects only provide `timeDerivative` which is the first derivative. 

Ok, I'd much rather see this done right, which is either: 

* adding a `secondTimeDerivative` to the `Function` class (and it's derived classes)
* having the user specify three (two?) function names for the derivatives then do the finite differencing in a new Function class; this is where the file reading code and differencing should go; or just derive from `PiecewiseBilinear` and add a parameter for the derivative order).

Sorry to be a such a pain here, but this is basically  an inflexible reimplementation of what the Function system was designed to be.